Fixed CarouselView Loop="False" rendering items incorrectly

kubaflo opened the pull request:
### Issues Fixed

Fixes https://github.com/dotnet/maui/issues/25192

|Before|After|
|--|--|
|<img src="https://github.com/user-attachments/assets/ec680fea-7b21-49eb-961d-9ef0eace0df9" width="300px"/>|<img src="https://github.com/user-attachments/assets/b830f4f3-5441-46c3-b9b9-fb9900974d4a" width="300px"/>|

kubaflo committed:
Fixed CarouselView Loop="False" rendering items incorrectly

kubaflo commented on src/Controls/src/Core/Handlers/Items/Android/MauiCarouselRecyclerView.cs:
```diff
@@ -571,46 +571,42 @@ void ClearLayoutListener()
 		protected override void OnMeasure(int widthMeasureSpec, int heightMeasureSpec)
 		{
-			if (Carousel.Loop)
+			// If the height or width are unbounded and the user is set to
```
I only removed the `if (Carousel.Loop)` condition but indentation changes make it harder to spot

jsuarezruiz commented:
/azp run

rmarinho commented:
/azp run

kubaflo committed:
Added snapshots

kubaflo commented on src/Controls/tests/TestCases.Shared.Tests/Tests/Issues/Issue25192.cs:
```diff
@@ -0,0 +1,23 @@
+        public void CarouselViewShouldRenderCorrectly()
+        {
+            App.WaitForElement("Item1");
+            VerifyScreenshot();
```
Done

jsuarezruiz commented:
/azp run

jsuarezruiz commented:
@rmarinho Thoughts?

rmarinho commented:
Seems this was failing UITests when rebased on main, so I added to revert.

The iOS seems not to render the next item, or the padding is different. 

![image](https://github.com/user-attachments/assets/e517fe87-2ffb-4f93-b014-12f36bb61c5b)

kubaflo commented:
> Seems this was failing UITests when rebased on main, so I added to revert.
> 
> The iOS seems not to render the next item, or the padding is different.
> 
> ![image](https://private-user-images.githubusercontent.com/1235097/387263773-e517fe87-2ffb-4f93-b014-12f36bb61c5b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzE5NDIzMjQsIm5iZiI6MTczMTk0MjAyNCwicGF0aCI6Ii8xMjM1MDk3LzM4NzI2Mzc3My1lNTE3ZmU4Ny0yZmZiLTRmOTMtYjAxNC0xMmYzNmJiNjFjNWIucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTExOCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDExMThUMTUwMDI0WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZmFiYjc0Y2NhYjg1NDU5ZDBkMDI3ZjExN2Q3YmJlN2VkYjk1M2U4YTVjYTEwMDA3NjhkMzRlZjY0NWRjM2FkMiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.eb3SrqXBdrjZMpQOv2JuxtKzpVmJZQ05cDmqdt-5RAE)

@rmarinho ah I see.
Item spacing was not working on iOS at the time of adding snapshot. The CarouselView2 for iOS fixed it, so I've updated the iOS screenshot to fix UI test
